src/utils/works.ts

```diff
@@ -7,16 +7,57 @@ import { getTagUrl } from "./tags";
 // correct type of page is passed to each method that extracts data.
 // Other than this, all pages are instances of CheerioAPI and can be used interchangeably.
 interface WorksPage extends CheerioAPI {
-  kind: "WorksPage";
+ kind: "WorksPage";
 }
+
+export interface WorkData {
+ workUrl: string;
+ workId: string;
+ chapterId: string | null;
+ collectionName: string | null;
+}
+
 const getWorksUrl = (tagName: string) => `${getTagUrl(tagName)}/works`;
 
 export const getWorksPage = async (tagName: string) => {
-  return cheerio.load(
-    (await axios.get<string>(getWorksUrl(tagName))).data
-  ) as WorksPage;
+ return cheerio.load(
+  (await axios.get<string>(getWorksUrl(tagName))).data
+ ) as WorksPage;
 };
 
 export const getTagId = ($worksPage: WorksPage) => {
-  return $worksPage(".rss")[0]?.attribs["href"].split("/")[2] || null;
+ return $worksPage(".rss")[0]?.attribs["href"].split("/")[2] || null;
+};
+
+interface WorkPage extends CheerioAPI {
+ kind: "WorkPage";
+}
+
+export const getWorkIdFromUrl = (url: string) => {
+ return url.match(/works\/(\d+)/)[1];
+};
+
+export const getChapterIdFromUrl = (url: string) => {
+ const chapterId = url.match(/chapters\/(\d+)/);
+ if (chapterId) return chapterId[1];
```

Comment:
You can do this more compactly with: `return chapterId?.[1]`.

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining